src/__tests__/retry.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Stops execution for {@link milliseconds} milliseconds.
+ */
+export async function milliseconds(milliseconds: number): Promise<void> {
+  await new Promise((_) => setTimeout(_, milliseconds));
+}
+
+/**
+ * Executes the function {@link fn}, retrying if it throws an error.
+ * Uses a linear strategy by retrying each {@link delayMs} milliseconds.
+ * If {@link maxAttempts} is reached, it throws the error returned by {@link fn} last execution.
+ */
+export async function retry(
+  fn: () => void,
+  maxAttempts: number = 40,
+  delayMs: number = 100,
+): Promise<void> {
+  let attempt = 0;
+  const execute = async (): Promise<void> => {
+    try {
+      return await fn();
+    } catch (error) {
+      if (attempt >= maxAttempts) {
+        throw error;
+      }
+    }
+
+    await milliseconds(delayMs);
+    attempt++;
+    return execute();
+  };
+  return execute();
+}
```

Comment:
Test-related utility which takes a function as a parameter, and executes this function repeatedly until success or until the attempts threshold is not reached. This can help to verify some piece of state that is not being mocked and is outside Jest's scope, like the asynchronous consumption of a RabbitMQ queue.

A simpler solution could be just using `setTimeout` but setting up the time lapse could be problematic, as a short one could lead to flaky tests under some conditions, and a longer one would impact the tests execution time.